include/zephyr/sensing/sensing.h

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) 2022-2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SENSING_H_
+#define ZEPHYR_INCLUDE_SENSING_H_
+
+/**
+ * @defgroup Sensing Subsystem APIs
+ * @defgroup sensing_api API
+ * @ingroup sensing
+ * @defgroup sensing_sensor_types Sensor Types
+ * @ingroup sensing
+ * @defgroup sensing_datatypes Data Types
+ * @ingroup sensing
+ */
+
+#include <zephyr/sensing/sensing_datatypes.h>
+#include <zephyr/sensing/sensing_sensor_types.h>
+
+/**
+ * @brief Sensing Subsystem API
+ * @addtogroup sensing_api
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @struct sensing_sensor_version
+ * @brief Sensor Version
+ */
+struct sensing_sensor_version {
+	union {
+		uint32_t value;
+		struct {
+			uint8_t major;
+			uint8_t minor;
+			uint8_t hotfix;
+			uint8_t build;
+		};
+	};
+};
+
+#define SENSING_SENSOR_INVALID_HANDLE (-1)
+#define SENSING_INDEX_ALL (-1)
+
+#define SENSING_SENSOR_VERSION(_major, _minor, _hotfix, _build)           \
+				(FIELD_PREP(GENMASK(31, 24), _major) |  \
+				 FIELD_PREP(GENMASK(23, 16), _minor) |  \
+				 FIELD_PREP(GENMASK(15, 8), _hotfix) |  \
+				 FIELD_PREP(GENMASK(7, 0), _build))
+
+
+/**
+ * @brief Sensor flag indicating if this sensor is on event reporting data.
+ *
+ * Reporting sensor data when the sensor event occurs, such as a motion detect sensor reporting
+ * a motion or motionless detected event.
+ */
+#define SENSING_SENSOR_FLAG_REPORT_ON_EVENT			BIT(0)
+
+/**
+ * @brief Sensor flag indicating if this sensor is on change reporting data.
+ *
+ * Reporting sensor data when the sensor data changes.
+ *
+ * Exclusive with  \ref SENSING_SENSOR_FLAG_REPORT_ON_EVENT
+ */
+#define SENSING_SENSOR_FLAG_REPORT_ON_CHANGE			BIT(1)
+
+
+/**
+ * @brief Sensor subsytem sensor state.
+ *
+ */
+enum sensing_sensor_state {
+	SENSING_SENSOR_STATE_NOT_READY = 1,
+	SENSING_SENSOR_STATE_READY = 2,
+};
+
+/**
+ * @brief Sensor subsystem sensor config attribute
+ *
+ */
+enum sensing_sensor_attribute {
+	SENSING_SENSOR_ATTRIBUTE_INTERVAL = 0,
+	SENSING_SENSOR_ATTRIBUTE_SENSITIVITY = 1,
+	SENSING_SENSOR_ATTRIBUTE_LATENCY = 2,
+	SENSING_SENSOR_ATTRIBUTE_MAX,
+};
+
+
+/**
+ * @brief Sensor data event receive callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer with sensor data.
+ *
+ * @param size The buffer size in bytes.
+ */
+typedef void (*sensing_data_event_t)(
+		int handle,
```

Comment:
Previously you had a different type for the handle, do we need to worry about `int` not necessarily being the same size always? Can we use `uint32_t` instead? Or better yet, hide it behind a typedef so we have the flexibility to change it later.